hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerTask.java

```diff
@@ -173,7 +173,9 @@ public void run() {
             TimeUnit.SECONDS);
         LOG.info("ContainerBalancer will sleep for {} seconds before starting" +
             " balancing.", delayDuration);
-        Thread.sleep(Duration.ofSeconds(delayDuration).toMillis());
+        if (isBalancerRunning()) {
+          Thread.sleep(Duration.ofSeconds(delayDuration).toMillis());
+        }
```

Comment:
I think the bug report is not about this sleep, rather the one in:

https://github.com/apache/ozone/blob/60fa8dd78d7f1b977e380928d87351ec5071e3fd/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerTask.java#L229-L237